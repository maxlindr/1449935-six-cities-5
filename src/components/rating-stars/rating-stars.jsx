import React from 'react';
import PropTypes from 'prop-types';

const MAX_RATING = 5;

export const RatingStarsType = {
  CARD: `place-card`,
  OFFER: `property`,
  REVIEW: `reviews`,
};

const convertRatingToPerc = (rating) => (rating / MAX_RATING) * 100;

const RatingStars = (props) => {
  const {rating, type} = props;

  return (
    <div className={`${type}__rating rating`}>
      <div className={`${type}__stars rating__stars`}>
        <span style={{width: `${convertRatingToPerc(rating)}%`}} />
        <span className="visually-hidden">Rating</span>
      </div>
      {
        type === RatingStarsType.OFFER
          ? <span className="property__rating-value rating__value">{rating}</span>
          : null
      }
    </div>
  );
};

RatingStars.propTypes = {
  rating: PropTypes.number.isRequired,
  type: PropTypes.oneOf([
    RatingStarsType.CARD,
    RatingStarsType.OFFER,
    RatingStarsType.REVIEW
  ]).isRequired,
};

export default React.memo(RatingStars);
