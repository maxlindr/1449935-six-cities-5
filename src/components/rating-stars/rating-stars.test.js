import React from 'react';
import renderer from 'react-test-renderer';
import {RatingStars, RatingStarsType} from './rating-stars';

test(`RatingStars должен корректно отображаться`, () => {
  const component = renderer.create(
      <RatingStars
        rating={4}
        type={RatingStarsType.OFFER}
      />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
