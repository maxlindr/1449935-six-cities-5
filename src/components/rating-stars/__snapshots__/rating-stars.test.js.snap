// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingStars должен корректно отображаться 1`] = `
<div
  className="property__rating rating"
>
  <div
    className="property__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "80%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
  <span
    className="property__rating-value rating__value"
  >
    4
  </span>
</div>
`;
